scripts/events_module/handle_short_events.py

```diff
@@ -406,10 +406,13 @@ def handle_mass_death(self):
             if self.main_cat not in self.dead_cats:
                 self.dead_cats.append(self.main_cat)  # got to include the cat that rolled for death in the first place
 
+            taken_cats = []
             for kitty in self.dead_cats:
                 if "lost" in self.chosen_event.tags:
                     kitty.gone()
-                    self.dead_cats.remove(kitty)
+                    taken_cats.append(kitty)
+            for kitty in taken_cats:
+                self.dead_cats.remove(kitty)
                 self.multi_cat.append(kitty)
```

Comment:
Nesting this and the next line within the `taken_cats` for loop means it will never happen for the cats who are just killed.  Both of those lines need to happen for all cats, regardless of lost or dead.

I'd suggest the removal loop should just be:
```
            for kitty in taken_cats:
                self.dead_cats.remove(kitty)
```
while the multi_cat appending and ID appending should happen in the `self.dead_cats` loop as it did before

Reply:
I updated it! I hope I understood correctly what you meant!